src/Symfony/Component/Translation/Formatter/IntlMessageFormatter.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Formatter;
+
+/**
+ * @author Guilherme Blanco <guilhermeblanco@hotmail.com>
+ * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
+ */
+class IntlMessageFormatter implements MessageFormatterInterface, ChoiceMessageFormatterInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function format($message, $locale, array $parameters = array())
+    {
+        $formatter = new \MessageFormatter($locale, $message);
```

Comment:
When the parsing of this message fails you get a very non-descriptive exception message (`Constructor failed`). Might be good for DX to rethrow a more descriptive one.